core/vm/evm.go

```diff
@@ -243,6 +245,7 @@ func (evm *EVM) DelegateCall(caller ContractRef, addr common.Address, input []by
 	ret, err = run(evm, snapshot, contract, input)
 	if err != nil {
 		evm.StateDB.RevertToSnapshot(snapshot)
+		contract.SetVmError(err)
```

Comment:
There's no reason to use such a mechanism, since the error is returned anyway from this function.